tests/cupy_tests/linalg_tests/test_product.py

```diff
@@ -256,14 +258,16 @@ def test_transposed_tensordot(self, xp, dtype):
         b = testing.shaped_arange((4, 3, 2), xp, dtype).transpose(2, 0, 1)
         return xp.tensordot(a, b)
 
-    @testing.for_all_dtypes()
+    @testing.for_all_dtypes(excludes=(numpy.uint8, numpy.int8, numpy.uint16,
```

Comment:
I'm not familiar with bytes handling, but using six to convert it to bytes looks safe for both python2 and 3. Can `encode('latin-1')` be enough?